plugin-api/src/main/java/org/sonar/api/batch/sensor/rule/AdHocRule.java

```diff
@@ -53,12 +56,29 @@ public interface AdHocRule {
 
   /**
    * Default severity of the rule.
+   * @deprecated since 10.1, use {@link #defaultImpacts()} instead
    */
+  @Deprecated(since = "10.1")
   Severity severity();
 
   /**
    * Type of the rule.
+   * @deprecated since 10.1, use {@link #defaultImpacts()} instead
    */
+  @Deprecated(since = "10.1")
   RuleType type();
 
+  /**
+   * Default impacts of the rule
+   * @since 10.1
+   */
+  Map<SoftwareQuality, org.sonar.api.issue.impact.Severity> defaultImpacts();
+
+  /**
+   * Clean Code Attribute of the rule according to Clean Code Taxonomy.
```

Comment:
We mention Clean Code Taxonomy in a few places in this PR. However, are the users of plugin-api aware what Clean Code Taxonomy entails?